### PR TITLE
Updated documentation

### PR DESCRIPTION
**Pull request by 0x5b62656e5d**

### CODE REVIEW

File: source/Docs/Software/SwerveModuleClass.rst [hunk 1]
```diff
@@ -1,3 +1,135 @@
...
-##################
\ No newline at end of file
+##################
+
+The SwerveModule class initializes the swerve module on the robot. The method ``getState`` returns a new object of the ``SwerveModuleState`` class, 
+allowing us to know the current state according to the linear speed of the throttle in meters per second and the degrees the encoder has rotated. The 
+method ``setState`` gets the current heading of the drive and takes a ``SwerveModuleState`` object. The method will use a PID controller to turn the 
+drive to the desired heading at a desired speed.
+
+Private Members
```
Comment (jazonshou) [inline source/Docs/Software/SwerveModuleClass.rst:10]: Call it "Instance Variables"

File: source/Docs/Software/SwerveModuleClass.rst [hunk 2]
```diff
@@ -1,3 +1,135 @@
...
+.. code-blocK:: java
+
+    mThrottle = new WPI_TalonFX(throttleID);
+    mRotor = new WPI_TalonFX(rotorID);
+    mRotorEncoder = new WPI_CANCoder(rotorEncoderID);
+
+Creates new objects of throttle, rotor, and encoder and assigns them to specific IDs that are the parameters from the constructor.
+
+Setting factory default
+========
```
[the remark this answers is not in the document]
Comment (jazonshou) [reply source/Docs/Software/SwerveModuleClass.rst:51]: Also another thing, the number of "=" or "*" or "-" needs to be equal to the number of characters in your subtitle

**Review by jazonshou — CHANGES_REQUESTED** [review]
Overall, very nice progress; keep up the good work!

File: source/Docs/Software/SwerveModuleClass.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (0x5b62656e5d) [reply source/Docs/Software/SwerveModuleClass.rst]: > After this, put all of the configuration in 1 code block, it'll make more sense soon (but basically different motors use different methods to initialize and we want to add different code for different motors... putting everything in 1 code block just makes it more concise for the reader)

Do I just put the whole config code in one block then depict it section by section?

[the remark this answers is not in the document]
Comment (jazonshou) [reply source/Docs/Software/SwerveModuleClass.rst]: Yeah, 1 block & comments

[the remark this answers is not in the document]
Comment (0x5b62656e5d) [reply source/Docs/Software/SwerveModuleClass.rst]: > Yeah, 1 block & comments in the code

something like this?
image

[the remark this answers is not in the document]
Comment (jazonshou) [reply source/Docs/Software/SwerveModuleClass.rst]: yes

[the remark this answers is not in the document]
Comment (0x5b62656e5d) [reply source/Docs/Software/SwerveModuleClass.rst]: Just curious why not do the documentation by blocks? (the current format i have rn, u can check I pushed to branch)

File: build/html/_sources/Docs/Software/SwerveModuleClass.rst.txt [hunk 3]
```diff
@@ -1,3 +1,189 @@
...
+
+The image below shows an example of a swerve module.
+
+.. figure:: swerveModule.jpg
+    :alt: swerveModule image
+
+    The blue circle shows the throttle; orange circle represents the rotor; pink circle represents the encoder
+
+Instantiate instance variables
+==============================
```
Comment (jazonshou) [inline build/html/_sources/Docs/Software/SwerveModuleClass.rst.txt:18]:
```suggestion
Instantiate Instance Variables
==============================

```

File: build/html/_sources/Docs/Software/SwerveModuleClass.rst.txt [hunk 4]
```diff
@@ -1,3 +1,189 @@
...
+    // throttle config
+    mThrottle.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+    mThrottle.configVoltageCompSaturation(Constants.kVoltageCompensation);
+    mThrottle.enableVoltageCompensation(true);
+    mThrottle.setNeutralMode(NeutralMode.Brake);
+
+1. Configures remote feedback device for motor controller
+2. Configures voltage compensation saturation (double)
+3. Toggles voltage compensation saturation (true/false)
+4. Sets neutral mode for throttle (Coast/Brake)
```
Comment (jazonshou) [inline build/html/_sources/Docs/Software/SwerveModuleClass.rst.txt:155]: everything here should be under the same code block

for explanations, just add comments into the code

for explaining the parameters of the constructor, just have a comment block

**Review by jazonshou — CHANGES_REQUESTED** [review]
good job with the changes, just a few more details

the reason why I want the entire constructor in 1 code block is that different types of electronics will use different methods in the constructor. this just makes it easier for me to add the different code later on.

File: source/Docs/Software/SwerveModuleClass.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (jazonshou) [reply source/Docs/Software/SwerveModuleClass.rst]: wdym by blocks

**Review by jazonshou — APPROVED** [review]
seems okay for now, ill make some changes

**Pull request merged**